Apsitvarkom.DataAccess/GoogleGeocoder.cs

```diff
@@ -22,15 +22,29 @@ public GoogleGeocoder(HttpClient httpClient, IApiKeyProvider apiKeyProvider)
         _httpClient = httpClient;
     }
 
-    public async Task<string?> ReverseGeocodeAsync(Coordinates coordinates)
+    public async Task<string?> ReverseGeocodeAsync(Coordinates coordinates, SupportedLanguages language)
     {
-        var query = $"json?latlng={coordinates.Latitude.ToString(CultureInfo.InvariantCulture)},{coordinates.Longitude.ToString(CultureInfo.InvariantCulture)}&language=lt&key={_apiKey}";
+        var query = $"json?latlng={coordinates.Latitude.ToString(CultureInfo.InvariantCulture)},{coordinates.Longitude.ToString(CultureInfo.InvariantCulture)}&language={Translated<string>.GetSupportedLocale(language)}&key={_apiKey}";
         var jsonStream = await _httpClient.GetStreamAsync(query);
         var response = await JsonSerializer.DeserializeAsync<ReverseGeocodingApiResponse>(jsonStream, SerializerOptions);
-
         return response?.Results?.FirstOrDefault()?.FormattedAddress;
     }
 
+    public async Task<Translated<string>> ReverseGeocodeAsync(Coordinates coordinates)
+    {
+        var title = new Translated<string>();
+        var languages = (SupportedLanguages[])Enum.GetValues(typeof(SupportedLanguages));
+
+        var tasks = languages.Select(language => ReverseGeocodeAsync(coordinates, language));
+        var geocodeResults = await Task.WhenAll(tasks);
+
+        var pairs = geocodeResults.Zip(languages, (name, language) => new { Language = language, Name = name ?? string.Empty });
```

Comment:
You could make the `Zip` before awaiting tasks (have them be a pair of `(SupportedLanguage, Task<string?>)` instead of `(SupportedLanguage,  string?)`) because we're now expecting the tasks to be in the same order of the languages in `SupportedLanguages`